tests/admintools.test.js

```diff
@@ -0,0 +1,213 @@
+
+      //var filesys = require('fs');
+      //var logfile = filesys.createWriteStream('out.txt', { flags: 'a'} )
+      
+      self.$.ajax = function mockAjax (url, opts) {
+        //logfile.write(url+'\n');
+        //console.log(url,opts);
+        if (opts && opts.success && opts.success.call) {
+          if (url.indexOf('/api/v1/treatments.json?&find[created_at][$gte]=')===0) {
+            url = '/api/v1/treatments.json?&find[created_at][$gte]='
+          }
+          if (url.indexOf('/api/v1/entries.json?&find[date][$gte]=')===0) {
+            url = '/api/v1/entries.json?&find[date][$gte]='
+          }
+          return {
+            done: function mockDone (fn) {
+                if (someData[url]) {
+                  console.log('+++++Data for ' + url + ' sent');
+                  opts.success(someData[url]);
+                } else {
+                  console.log('-----Data for ' + url + ' missing');
+                  opts.success([]);
+                }
+              fn();
+              return self.$.ajax();
+            },
+            fail: function mockFail (fn) {
+              fn({status: 400});
+              return self.$.ajax();
+            }
+          };
+        }
+        if (url && url.method === 'DELETE') {
+          return {
+            done: function mockDone (fn) {
+              fn();
+              return {
+                fail: function mockFail () {
+                }
+              };
+            }
+          };
+        }
+      };
+
+      self.$.plot = function mockPlot () {
+      };
+
+      var d3 = require('d3');
+      //disable all d3 transitions so most of the other code can run with jsdom
+      d3.timer = function mockTimer() { };
+
+      benv.expose({
+        $: self.$
+        , jQuery: self.$
+        , d3: d3
+        , serverSettings: serverSettings
+        , io: {
+          connect: function mockConnect ( ) {
+            return {
+              on: function mockOn ( ) { }
+            };
+          }
+        }
+      });
+
+      benv.require(__dirname + '/../bundle/bundle.source.js');
+      benv.require(__dirname + '/../static/admin/js/admin.js');
+
+      done();
+    });
+  });
+
+  after(function (done) {
+    benv.teardown(true);
+    done();
+  });
+
+  it ('should produce some html', function (done) {
+    var plugins = require('../lib/plugins/')().registerClientDefaults();
+    var client = require('../lib/client');
+
+    var hashauth = require('../lib/hashauth');
+    hashauth.init(client,$);
+    hashauth.verifyAuthentication = function mockVerifyAuthentication(next) {
+      hashauth.authenticated = true;
+      next(true);
+    };
+
+     window.confirm = function mockConfirm (text) {
+       console.log('Confirm:', text);
+       return true;
+     };
+
+     window.alert = function mockAlert () {
+       return true;
+     };
+
+    client.init(serverSettings, plugins);
+    client.dataUpdate(nowData);
+    
+    var result = $('body').html();
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['result' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1026365146/issues/source?bid=2452379&fileBranchId=2878156#l185)